### DISCUSSION TITLE
Recommended IDE for Publii development

### DISCUSSION THREAD

**Discussion by johnwoltman** [GetPublii/Publii#1757] [General]
Does anyone, maybe even someone on the Publii team, have a recommendation for an IDE for Publii development? I come from a C++/Python background where I use Qt Creator for the C++ stuff and PyCharm for Python stuff.  Not that familiar with Javascript options. Thanks!

**Comment by bjazmoore** [comment]
What sort of development?  VSCode is nice.  I also like Atom even though it has been sunsetted.

**Reply by johnwoltman** [reply]
In particular developing Publii.  I haven't tried VSCode but I hear good things.

**Reply by johnwoltman** [reply]
To add some detail: I'm looking for an IDE that supports autocomplete, and a debugger.  I don't even know what debugging a node app looks like, but I've always used debuggers for other languages.

**Reply by riker09** [reply]
You should definitely give VScode a try, then. It features a debugger for many languages, not sure if it can be used with developing a static website, though. What you can do for sure is using a browser-integrated debugger. I know for sure that Edge and Firefox have on on board, Chrome _should_ feature one as well.

**Reply by candidexmedia** [reply]
@johnwoltman are you looking to develop the Publii GUI app itself, or develop a Publii theme?

**Reply by johnwoltman** [reply]
I plan on developing Publii itself.
